bakerydemo/blog/models.py

```diff
@@ -138,10 +138,29 @@ def tag_archive(self, request, tag=None):
                 messages.add_message(request, messages.INFO, msg)
             return redirect(self.url)
 
-        blogs = BlogPage.objects.filter(tags=tag).live().descendant_of(self)
+        blogs = self.blogs(tag=tag)
 
         context = {
             'tag': tag,
             'blogs': blogs
         }
         return render(request, 'blog/blog_index_page.html', context)
+
+    def blogs(self, tag=None):
```

Comment:
Not a blocker, but I would probably call this `get_posts()` (preferable to use `get_` for anything that retrieves other data, and distinguishing between a blog, which is the whole thing, vs a post, i.e. an individual post in a blog).